### PR TITLE
fix(core/structure): support large levels in TOC

### PR DESCRIPTION
**Pull request by thijsbrentjens**
See discussion in Geonovum repo: https://github.com/Geonovum/respec/pull/185

Calculating the level for the TOC failed when large (>=10) number of sections where available.

### CODE REVIEW

**Comment by sidvishnoi** [comment]
Not sure if we should test this or it's fine. cc @marcoscaceres

**Comment by marcoscaceres** [comment]
Wouldn't hurt to add a test... catch regressions in the future.

File: tests/spec/core/structure-spec.js [hunk 1]
```diff
@@ -172,6 +172,25 @@ describe("Core - Structure", () => {
     );
   });
 
+  it("should correctly put all headings until maxTocLevel in ToC", async () => {
+    const times = (n, fn) =>
+      Array.from({ length: n }, (_, i) => fn(i)).join("\n");
+    const body = `
+      ${makeDefaultBody()}
+      ${times(9, () => `<section><h2>pass</h2></section>`)}
```
Comment (marcoscaceres) [inline tests/spec/core/structure-spec.js:180]: Might be good to add comment here why the `9` and `10` different sections are needed. 

Also, as we are checking maxTocLevel, would we want to go to level 3 and check level 3 stuff is not actually there?

Comment (sidvishnoi) [reply tests/spec/core/structure-spec.js:180]: > would we want to go to level 3 and check level 3 stuff is not actually there?

That's tested under `should limit ToC depth with maxTocLevel`

> Might be good to add comment here why the 9 and 10 different sections are needed.

:+1:

**Pull request merged**